sandpack-react/src/contexts/sandpackContext.tsx

```diff
@@ -406,12 +436,37 @@ class SandpackProvider extends React.PureComponent<
             delete this.unsubscribeQueuedListeners[clientId][listenerId];
           }
         };
+
+        return unsubscribeListener;
       }
     } else {
-      const unsubs = Object.values(this.clients).map((client) =>
+      // Push to the **global** queue
+      const listenerId = generateRandomId();
+      this.queuedListeners.global[listenerId] = listener;
+
+      // Add to the current clients
+      const clients = Object.values(this.clients);
+      const currentClientListeners = clients.map((client) =>
         client.listen(listener)
       );
-      return () => unsubs.forEach((unsub) => unsub());
+      const currentClientListenersUnsubscribe = () => {
```

Comment:
nitpick: because there are already a bunch of functions/variables here with similar names, I would keep this as an anonymous function at line 466, because it is similar with the other foreach that unsubscribes from each "future" client. Also, `current` might be misleading in the name, since timeline is super complex in this function :)